# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieCommons.cpp
# Commit: fe6e63a2f914
# Full Hash: fe6e63a2f9141d9eeec41bfde52cc34b33c170fe
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-10-09 14:54 +0000
# Description:
#   Bug 1992536 - Use cookie creation time instead or PR_Now() to validate the expiration time, r=valentin,cookie-reviewers
# ==============================================================================

diff -r 0147060f88c2 -r fe6e63a2f914 netwerk/cookie/CookieCommons.cpp
--- a/netwerk/cookie/CookieCommons.cpp	Thu Oct 09 14:36:53 2025 +0000
+++ b/netwerk/cookie/CookieCommons.cpp	Thu Oct 09 14:54:32 2025 +0000
@@ -377,7 +377,7 @@
   aCookieParser.Parse(baseDomain, requireHostMatch, cookieStatus, cookieString,
                       EmptyCString(), false, isForeignAndNotAddon,
                       mustBePartitioned, aDocument->IsInPrivateBrowsing(),
-                      on3pcbException, PR_Now() / PR_USEC_PER_MSEC);
+                      on3pcbException, PR_Now());
 
   if (!aCookieParser.ContainsCookie()) {
     return nullptr;
@@ -1023,17 +1023,17 @@
 }
 
 // static
-int64_t CookieCommons::GetCurrentTimeFromChannel(nsIChannel* aChannel) {
+int64_t CookieCommons::GetCurrentTimeInUSecFromChannel(nsIChannel* aChannel) {
   nsCOMPtr<nsITimedChannel> timedChannel = do_QueryInterface(aChannel);
   if (timedChannel) {
     PRTime currentTimeInUSec = 0;
     nsresult rv = timedChannel->GetResponseStartTime(&currentTimeInUSec);
     if (NS_SUCCEEDED(rv) && currentTimeInUSec) {
-      return currentTimeInUSec / PR_USEC_PER_MSEC;
+      return currentTimeInUSec;
     }
   }
 
-  return PR_Now() / PR_USEC_PER_MSEC;
+  return PR_Now();
 }
 
 }  // namespace net